x-pack/plugins/ml/public/application/datavisualizer/data_comparison/data_drift_index_patterns_editor.tsx

```diff
@@ -0,0 +1,370 @@
+
+    let dataView;
+
+    if (!foundDataViewId) {
+      const modifiedDataViewName =
+        dataViewName === '' ? `${indicesName}-${timeFieldName}` : dataViewName;
+      if (canEditDataView && createAdHocDV === false) {
+        dataView = await dataViews.createAndSave({
+          title: indicesName,
+          name: modifiedDataViewName,
+          timeFieldName,
+        });
+      } else {
+        dataView = await dataViews.create({
+          title: indicesName,
+          name: modifiedDataViewName,
+          timeFieldName,
+        });
+      }
+    }
+    const dataViewId = foundDataViewId ?? dataView?.id;
+    const url = await locator.getUrl({
+      page: ML_PAGES.DATA_COMPARISON,
+      pageState: {
+        index: dataViewId,
+        reference: referenceIndexPattern,
+        production: productionIndexPattern,
+        timeFieldName,
+      },
+    });
+
+    await navigateToPath(url);
+  };
+
+  const firstSetOfSteps = [
+    {
+      title: 'Pick index pattern for reference data indices',
+      children: (
+        <EuiFlexItem grow={false}>
+          <DataViewEditor
+            key={'reference'}
+            label={
+              <FormattedMessage
+                id="xpack.ml.dataDrift.indexPatternsEditor.referenceData"
+                defaultMessage="Index pattern for reference data indices"
+              />
+            }
+            dataViewEditorService={referenceDataViewEditorService}
+            indexPattern={referenceIndexPattern}
+            setIndexPattern={setReferenceIndexPattern}
+          />
+        </EuiFlexItem>
+      ),
+    },
+    {
+      title: 'Pick index pattern for comparison data indices',
+      children: (
+        <EuiFlexItem grow={false}>
+          <DataViewEditor
+            key={'comparison'}
+            label={
+              <FormattedMessage
+                id="xpack.ml.dataDrift.indexPatternsEditor.comparisonData"
+                defaultMessage="Index pattern for comparison data indices"
+              />
+            }
+            dataViewEditorService={productionDataViewEditorService}
+            indexPattern={productionIndexPattern}
+            setIndexPattern={setProductionIndexPattern}
+          />
+        </EuiFlexItem>
+      ),
+    },
+    {
+      title: 'Additional settings',
+      children: (
+        <EuiFlexGroup direction="column">
+          {combinedTimeFieldOptions.length > 0 ? (
+            <EuiFormRow
+              label={i18n.translate('xpack.ml.dataDrift.indexPatternsEditor.timeField', {
+                defaultMessage: 'Time field',
+              })}
+              fullWidth
+              color={'disabled'}
+            >
+              <>
+                <EuiComboBox<string>
+                  placeholder={i18n.translate(
+                    'xpack.ml.dataDrift.indexPatternsEditor.timestampField',
+                    {
+                      defaultMessage: 'Select an optional timestamp field',
```

Comment:
Could we use the same UX as in the Data view creation page here, as otherwise it is quite easy to unintentionally skip over the step where you select the timestamp field. The Data view page defaults to the time field where only one exists, and forces the user to select the option if they don't want to use a time field.

<img width="590" alt="image" src="https://github.com/elastic/kibana/assets/7405507/20135c1b-fb87-4a8f-a820-b5ffa135aec8">